jwst/datamodels/schema.py

```diff
@@ -6,6 +6,7 @@
 from asdf import AsdfFile
 from asdf import schema as asdf_schema
 
+from .extension import BaseExtension
```

Comment:
Is `BaseExtension` used anywhere?

Reply:
No, it is not.  Thanks for catching this.  Also, the other two extensions should probably not be used here either, as that is the old way of doing things.  Gosh this codebase sprawls.  I will update this PR.

Reply:
Yes, I can delete the whole `read_schema()` function.  It is no longer used now that we use `asdf.schema.load_schema()`.  This was probably an oversight in the past.